scripts/beekeeper.pl

```diff
@@ -507,6 +514,47 @@ sub register_beekeeper {
     return $beekeeper;
 }
 
+
+sub big_red_button {
+    my ( $self, $valley ) = @_;
+
+    my $bk_a = $self->{dba}->get_BeekeeperAdaptor();
+    my $blocked_beekeepers;
+
+    # Save a list of IDs of beekeepers which were blocked earlier so
+    # that we can not mention them while reporting the current blocking.
+    $blocked_beekeepers = $bk_a->fetch_all( 'is_blocked = 1' );
+    my %previously_blocked_ids;
+    while ( my $blocked_bk = shift @{ $blocked_beekeepers } ) {
+        $previously_blocked_ids{ $blocked_bk->dbID() } = 1;
+    }
+
+    # Begin the shutdown by blocking all registered beekeepers so that
+    # none of them start spawning new workers just as this one tries to
+    # kill all workers.
+    $bk_a->block_all_alive_beekeepers();
```

Comment:
This is a bit awkward because it will block the red-button beekeeper itself, even though we know it's going to exit right after